xbmc/profiles/dialogs/GUIDialogLockSettings.cpp

```diff
@@ -75,6 +76,10 @@ bool CGUIDialogLockSettings::ShowAndGetLock(CProfile::CLock &locks, int buttonLa
     return false;
 
   locks = dialog->m_locks;
+
+  // refresh favourites due to possible visibility changes
```

Comment:
The comment should contain why this needs to be called here. Not a general reason but the specific reason.